### PR TITLE
Convert some uses of coll.toArray(coll.size()) to coll.toArray(Type[]::new)

### PR DESCRIPTION
**Pull request by thecoop**
It's not recommended to use `toArray(coll.size())`, use `toArray(Type[]::new)` instead.

Also tidy up some code in the schedule cron types.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/XContentTestUtils.java [hunk 1]
```diff
@@ -268,7 +270,7 @@ public static BytesReference insertRandomFields(
      *  <li>"foo3.foo4</li>
      * </ul>
      */
-    static List<String> getInsertPaths(XContentParser parser, Stack<String> currentPath) throws IOException {
+    static List<String> getInsertPaths(XContentParser parser, Deque<String> currentPath) throws IOException {
```
Comment (thecoop) [inline test/framework/src/main/java/org/elasticsearch/test/XContentTestUtils.java:273]: Converting away from class with synchronized methods

File: x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/YearlySchedule.java [hunk 2]
```diff
@@ -62,11 +63,10 @@ public static Builder builder() {
...
     static String[] crons(YearTimes[] times) {
         assert times.length > 0 : "at least one time must be defined";
-        Set<String> crons = Sets.newHashSetWithExpectedSize(times.length);
-        for (YearTimes time : times) {
-            crons.addAll(time.crons());
-        }
-        return crons.toArray(new String[crons.size()]);
+        return Arrays.stream(times)
+            .flatMap(yt -> yt.crons().stream())
+            .distinct()
```
Comment (thecoop) [inline x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/trigger/schedule/YearlySchedule.java:68]: This one uses a set, the other schedule types use list...?

**Review by rjernst — APPROVED** [review]
LGTM, with the caveat of separating unrelated changes like Stack -> Deque to a separate PR.

File: test/framework/src/main/java/org/elasticsearch/test/XContentTestUtils.java (hunk 1 above)
Comment (thecoop) [reply test/framework/src/main/java/org/elasticsearch/test/XContentTestUtils.java]: #95278

**Pull request merged**